tools/execsnoop.py

```diff
@@ -70,42 +83,54 @@
     const char __user *const __user *__argv,
     const char __user *const __user *__envp)
 {
-    char fname[ARGSIZE] = {};
-    bpf_probe_read(&fname, sizeof(fname), (void *)(filename));
-    bpf_trace_printk("ARG %s\\n", fname);
+    // create data here and pass to submit_arg to save space on the stack (#555)
+    struct data_t data = {};
+    data.pid = bpf_get_current_pid_tgid() >> 32;
+    bpf_get_current_comm(&data.comm, sizeof(data.comm));
+    data.type = EVENT_ARG;
 
-    int i = 1;  // skip first arg, as we printed fname
+    submit_arg(ctx, (void *)&filename, &data);
```

Comment:
can we split submit_arg() into two functions, so we don't have to do '&filename' here?
since extra probe_read is not needed in this case and makes the code confusing to read.
Like
int submit_arg(ctx, ptr, data) {
 const char *argp;
bpf_probe_read(&argp, sizeof(argp), ptr);
if (argp)
 return __submit_arg(ctx, argp, data);
}
and call __ version here ?